tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -300,7 +300,10 @@ port::Status GetLoadedCudnnVersion(CudnnVersion* version) {
 
 }  // namespace
 
-CudnnSupport::CudnnSupport(CUDAExecutor* parent) : parent_(parent) {}
+CudnnSupport::CudnnSupport(CUDAExecutor* parent) : parent_(parent) {
+  tensorflow::ReadBoolFromEnvVar("TF_CUDNN_DETERMINISTIC", false,
```

Comment:
Since it's only for convolution, maybe renamed it to TF_CUDNN_DETERMINISTIC_CONV, and cudnn_deterministic_conv_?

Reply:
I'm happy with that if it's preferable to others. I chose the more generic name because the intention would be to incrementally enhance its effect so that it makes all cuDNN functionality deterministic. The longer-term goal is to switch all of these modes on with a single ConfigProto bool field (e.g. `deterministic`), which may include functionality beyond cuDNN.

Reply:
Ah, that intention is fine, and TF_CUDNN_DETERMINISTIC is the right name for it. In this case, can you add a TODO(cudnn), saying that this env var needs to support all APIs in cudnn, but currently only supports conv?

Reply:
Will do. I'll add that above where the env var is read in the constructor.

Reply:
This is done. I added the number of this PR, rather than cudnn (which seems ambiguous), so that this conversation can be referenced. Is that acceptable?

Reply:
I think `TODO(pr/24355): ...` is better. Otherwise LGTM.

Reply:
Sorry, I wasn't clear about it, but I was asking for a change from `TODO(PR #24355)` to `TODO(pr/24355)`. The latter is more consistent with other TODOs.

Reply:
Got it. I'm making that change now.

Reply:
Please could someone tell me what "Google internal checks FAILED" means? I cannot see the output from that.

Reply:
Awesome. Thanks, Tim. I'll take account of this when I re-open this request.